### ISSUE TITLE
Tune temporal batching for model optimization

### ISSUE THREAD

**Issue by JordiBolibar** [ODINN-SciML/MassBalanceMachine#12]
This could be a key element in model training, since it acts as a hyperparameter. 

Basically, we have two main strategies:

- Whenever we integrate and hit a seasonal period, we can add a new tree or backpropagate. And whenever we hit an annual period, we do the same. This means, to optimize the model for the smallest temporal period possible.
- Alternatively, we can do this in conjuction, and only optimize whenever we have integrated both the winter and summer seasonal data + the annual value. We compute a combined loss of these 3 values before optimizing the model.
